.github/workflows/ci.yaml

```diff
@@ -88,12 +91,20 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Set channel
+        run: |
+          juju_channel=$(echo "${{ matrix.agent-versions }}" | cut -c 1-3)
+          echo "channel=${juju_channel}/stable" >> "$GITHUB_ENV"
+          juju_major=$(echo "${{ matrix.agent-versions }}" | cut -c 1)
+          echo "libjuju=juju${juju_major}" >> "$GITHUB_ENV"
       - name: Setup operator environment
         # TODO: Replace with custom image on self-hosted runner
         uses: charmed-kubernetes/actions-operator@main
         with:
           provider: microk8s
-          bootstrap-options: "--agent-version ${{ matrix.agents-version }}"
+          channel: "1.27-strict/stable"
```

Comment:
Actions operator needs strictly confined microk8s to setup juju 3. I don't think we can track anything in renovate, so we'll need to move the pin manually for the time being.